api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxNewCohortQueries.java

```diff
@@ -0,0 +1,204 @@
+		            + tbMetadata.getTUBERCULOSIS_TREATMENT_PLANConcept().getConceptId() + " and o.value_coded="
+		            + hivMetadata.getYesConcept().getConceptId()
+		            + " and e.encounter_datetime >= :onOrAfter and e.encounter_datetime <=:onOrBefore and e.location_id=:location group by p.patient_id");
+		notifiedToBeOnTbTreatment.addParameter(new Parameter("onOrAfter", "onOrAfter", Date.class));
+		notifiedToBeOnTbTreatment.addParameter(new Parameter("onOrBefore", "onOrBefore", Date.class));
+		notifiedToBeOnTbTreatment.addParameter(new Parameter("location", "location", Location.class));
+		return notifiedToBeOnTbTreatment;
+	}
+	
+	/**
+	 * Build TxNew Cohort Queries
+	 * 
+	 * @return TxNew CompositionCohortDefinition
+	 */
+	@DocumentedDefinition(value = "TxNew")
+	public CohortDefinition getTxNewCohort(CohortDefinition inARTProgramDuringTimePeriod, CohortDefinition patientWithSTARTDRUGSObs,
+	        CohortDefinition patientWithHistoricalDrugStartDateObs, CohortDefinition patientsWithDrugPickUpEncounters,
+	        CohortDefinition transferredFromOtherHealthFacility, CohortDefinition AgeCohort, CohortDefinition GenderCohort) {
+		CompositionCohortDefinition TxNewForAge = new CompositionCohortDefinition();
+		TxNewForAge.setName("patientBelow1YearEnrolledInHIVStartedART");
+		TxNewForAge.addParameter(new Parameter("onOrAfter", "onOrAfter", Date.class));
+		TxNewForAge.addParameter(new Parameter("onOrBefore", "onOrBefore", Date.class));
+		TxNewForAge.addParameter(new Parameter("location", "location", Location.class));
+		TxNewForAge.addParameter(new Parameter("effectiveDate", "effectiveDate", Date.class));
+		TxNewForAge.getSearches().put("1", new Mapped<CohortDefinition>(inARTProgramDuringTimePeriod,
+		        ParameterizableUtil.createParameterMappings("onOrAfter=${onOrAfter},onOrBefore=${onOrBefore},location=${location}")));
+		TxNewForAge.getSearches().put("2", new Mapped<CohortDefinition>(patientWithSTARTDRUGSObs,
+		        ParameterizableUtil.createParameterMappings("onOrAfter=${onOrAfter},onOrBefore=${onOrBefore},location=${location}")));
+		TxNewForAge.getSearches().put("3", new Mapped<CohortDefinition>(patientWithHistoricalDrugStartDateObs,
+		        ParameterizableUtil.createParameterMappings("onOrAfter=${onOrAfter},onOrBefore=${onOrBefore},location=${location}")));
+		
+		TxNewForAge.getSearches().put("4", new Mapped<CohortDefinition>(patientsWithDrugPickUpEncounters,
+		        ParameterizableUtil.createParameterMappings("onOrAfter=${onOrAfter},onOrBefore=${onOrBefore},location=${location}")));
+		
+		TxNewForAge.getSearches().put("5", new Mapped<CohortDefinition>(transferredFromOtherHealthFacility,
+		        ParameterizableUtil.createParameterMappings("onOrAfter=${onOrAfter},onOrBefore=${onOrBefore},location=${location}")));
+		
+		String compositionString = "((1 AND 2 AND 3 AND 4) AND (NOT 5))";
+		
+		if (AgeCohort != null) {
```

Comment:
I don't like this, I think we can do this better, but this is the first step towards improving how the cohorts are grouped.

Reply:
Sure, we can rewrite this better